### DISCUSSION TITLE
cannot import name 'constant_init' from 'mmcv.cnn'

### DISCUSSION THREAD

**Discussion by Berry-Wu** [open-mmlab/mmcv#2912] [Q&A]
my version is：
Package    Version    Source
---------  ---------  --------------------------------------
mmcv       2.0.1      https://github.com/open-mmlab/mmcv
mmengine   0.8.4      https://github.com/open-mmlab/mmengine
mmpose     1.1.0      https://github.com/open-mmlab/mmpose
And I find that there is no `constant_init` in mmcv/cnn. What should I modify?

**Comment by sidiangongyuan** [comment]
I have the same question about this. Maybe it's a problem about the version. Have you solved it?

**Comment by matanPercepto** [comment]
I have the same issue, probably a change made by the move to mmcv 2, but can't find where it was moved to. 
Any luck?

**Reply by sidiangongyuan** [reply]
Yes, I have solved this problem. I just change version of my mmcv.   Some package has been moved to mmengine

**Reply by yzrs** [reply]
Excuse me,could please tell me the version of mmcv that you have installed?Thank you.

**Reply by sidiangongyuan** [reply]
> Excuse me,could please tell me the version of mmcv that you have installed?Thank you.

mmcv-full                 1.4.0                    pypi_0    pypi
mmdet                     2.19.1                   pypi_0    pypi
mmdet3d                   1.0.0rc0                  dev_0    <develop>
mmsegmentation            0.20.2                   pypi_0    pypi

**Comment by chengamo** [comment]
This will solve the problem "from mmengine.model import (constant_init, kaiming_init)"
